### PR TITLE
testapi: Try to return sane default in non-qemu host_ip calls

### PR DESCRIPTION
**Pull request by okurz**
In case of non-qemu tests rely on host_ip, e.g. for log uploading. For
this so far a valid WORKER_HOSTNAME would need to be provided. We can
provide a sane default what is supplied by "hostname -f" which should
return a valid FQDN but can still be overridden with the variable
WORKER_HOSTNAME as necessary.

Related progress issue: https://progress.opensuse.org/issues/120261

### CODE REVIEW

File: testapi.pm [hunk 1]
```diff
@@ -2091,12 +2091,15 @@ sub diag (@args) { bmwqemu::diag(@args) }
...
+Optional named parameter C<inside_sut> in C<$args> will force using actual
+worker IP/hostname even on KVM guests if set to 0.
+
+In case of non-qemu tries to return a sane default if the test variable
+C<WORKER_HOSTNAME> was not specified.
 
 =cut
 
-sub host_ip ($args = {}) { (check_var('BACKEND', 'qemu') && ($args->{inside_sut} // 1)) ? get_var('QEMU_HOST_IP', '10.0.2.2') : get_required_var('WORKER_HOSTNAME') }
+sub host_ip ($args = {}) { (check_var('BACKEND', 'qemu') && ($args->{inside_sut} // 1)) ? get_var('QEMU_HOST_IP', '10.0.2.2') : get_var('WORKER_HOSTNAME', qx{hostname -f}) }
```
[the remark this answers is not in the document]
Comment (okurz) [reply testapi.pm]: yep, done the second

**Comment by Martchus** [comment]
You could actually make use of `Net::Domain` which is a standard Perl module (see https://github.com/os-autoinst/openQA/pull/4900/files for an example). I've already tested the `hostfqdn` function on multiple production worker hosts and it works as expected.

File: testapi.pm [hunk 2]
```diff
@@ -2091,12 +2091,15 @@ sub diag (@args) { bmwqemu::diag(@args) }
...
+Optional named parameter C<inside_sut> in C<$args> will force using actual
+worker IP/hostname even on KVM guests if set to 0.
+
+In case of non-qemu tries to return a sane default if the test variable
+C<WORKER_HOSTNAME> was not specified.
 
 =cut
 
-sub host_ip ($args = {}) { (check_var('BACKEND', 'qemu') && ($args->{inside_sut} // 1)) ? get_var('QEMU_HOST_IP', '10.0.2.2') : get_required_var('WORKER_HOSTNAME') }
+sub host_ip ($args = {}) { (check_var('BACKEND', 'qemu') && ($args->{inside_sut} // 1)) ? get_var('QEMU_HOST_IP', '10.0.2.2') : $bmwqemu::vars{WORKER_HOSTNAME} //= qx{hostname -f} }
```
[the remark this answers is not in the document]
Comment (okurz) [reply testapi.pm]: > You could actually make use of `Net::Domain` which is a standard Perl module (see https://github.com/os-autoinst/openQA/pull/4900/files for an example). I've already tested the `hostfqdn` function on multiple production worker hosts and it works as expected.

Done, see second commit

**Comment by okurz** [comment]
> You could actually make use of `Net::Domain` which is a standard Perl module (see https://github.com/os-autoinst/openQA/pull/4900/files for an example). I've already tested the `hostfqdn` function on multiple production worker hosts and it works as expected.

Done, see second commit

File: cpanfile [hunk 3]
```diff
@@ -47,6 +47,7 @@ requires 'Mojo::UserAgent';
 requires 'Mojolicious', '>= 8.42';
 requires 'Mojolicious::Lite';
 requires 'Net::DBus';
+requires 'Net::Domain';
```
Comment (Martchus) [inline cpanfile:50]: With "standard Perl module" I actually mean *core* Perl module. So there's no need to specify the module here (but supposedly also no harm in doing so).

**Review by Martchus — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Pull request merged**